aeron-driver/src/main/java/io/aeron/driver/DriverConductor.java

```diff
@@ -1373,6 +1447,67 @@ private ArrayList<SubscriberPosition> createSubscriberPositions(
         return subscriberPositions;
     }
 
+    private void executeAsyncClientTask(
```

Comment:
I think the naming of this function should reflect that is is performing the `UdpChannel::parse`.  I think this could be generalised as well, if the AsyncResult carried both a success result and a failure result.  Then the result will be forwarded back to the command queue to let the caller take the appropriate action based on the success or failure rather than routing it directly back to the client from here.  With this approach we could probably fold the functionality of this method with `executeAsyncTask`.

Reply:
@mikeb01 I went this route originally and had `UdpChannel` parsing be passed as `Supplier<UdpChannel>` parameter. I then realized that all client tasks always parse the channel and decided to avoid allocating a `Supplier` instance just for that.
_Btw, the `AsyncResult` already carries both ok and error result._

On your second point: the reason why the `executeAsyncClientTask` is not calling `executeAsyncTask` is, because we want to limit number of client actions to one in-flight but allow any number of other async tasks. To that end the `executeAsyncClientTask` is setting `asyncClientCommandInFlight = true` and is responsible for resetting this flag when the action completes.